projects/components/src/table/data/table-cdk-data-source.ts

```diff
@@ -284,14 +301,23 @@ export class TableCdkDataSource implements DataSource<TableRow> {
 
     const request = this.buildRequest(columnConfigs, pageEvent, filters, queryProperties);
 
-    return this.hasCachedRowsForRequest(request) ? this.fetchCachedData(request) : this.fetchNewData(request);
+    return this.hasCacheForRequest(request) ? this.fetchCachedData(request) : this.fetchNewData(request);
+  }
+
+  private haveColumConfigsChanged(request: TableDataRequest): boolean {
+    if (isNil(this.cachedData.request)) {
+      return true;
+    }
+
+    return !isEqual(request.columns, this.cachedData.request.columns);
```

Comment:
if we don't change the object type, wouldn't this condition still do a valid check?

Reply:
comment below addresses this